File: 1.two-sum.cs
/*
 * @lc app=leetcode id=1 lang=csharp
 *
 * [1] Two Sum
 */

// @lc code=start
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        //Create a hash table with key as target searching and value as index matched
        Hashtable dic = new Hashtable();

        for(int i = 0; i < nums.Length; i++){
            if(dic.ContainsKey(nums[i]))
                return new int[] { (int)dic[nums[i]], i };
            else
                dic[target - nums[i]] = i;
        }

        return new int[] { };
    }
}
// @lc code=end

File: 1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let dict = {};
  for (var i = 0; i < nums.length; i++) {
    if (dict.hasOwnProperty(nums[i])) return [i, dict[nums[i]]];
    else dict[target - nums[i]] = i;
  }
};
// @lc code=end
